add post route documentation (#416)
meilisearch/documentation: references/search.md

# Search

> Meilisearch exposes 2 routes to perform searches:

Reviewer: ```suggestion
MeiliSearch exposes 2 routes to perform searches:
```

- A POST route: this is the preferred route when using API authentication, as it allows preflight request cache and better performances.

> - A GET route: the usage of this route is discouraged, unless you have good reason to do otherwise (specific caching abilities for example)

Reviewer 2: ```suggestion
- A GET route: the usage of this route is discouraged, unless you have good reason to do otherwise (specific caching abilities for example).
```

> Other than the differences mentioned above, the two route are strictly equivalent.

Reviewer 2: ```suggestion
Other than the differences mentioned above, the two routes are strictly equivalent.
```

## Search in an index with POST route

<RouteHighlighter method="POST" route="/indexes/:index_uid/search"/>

Search for documents matching a specific query in the given index.

> This is the preferred route to perform search when an API key is required, as is allows for preflight queries to be cached.

Reviewer 2: ```suggestion
This is the preferred route to perform search when an API key is required, as it allows for [preflight requests](https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request) to be cached. Caching preflight requests **improves considerably the speed of the search**.
```

# Search

Meilisearch exposes 2 routes to perform searches:

> - A POST route: this is the preferred route when using API authentication, as it allows [preflight request](https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request) cache and better performances.

Reviewer 3: Maybe 'caching'.

```suggestion
- A POST route: this is the preferred route when using API authentication, as it allows [preflight request](https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request) caching and better performances.
```

- A GET route: the usage of this route is discouraged, unless you have good reason to do otherwise (specific caching abilities for example)

> Other than the differences mentioned above, the two route are strictly equivalent.

Reviewer 3: ```suggestion
Other than the differences mentioned above, the two routes are strictly equivalent.
```

| **attributestToCrop**     | `[Strings]`            | Attributes whose values have to be cropped                                                      | `true`                                                                                            |
| **cropLength**            | `Integer`              | Length used to crop field values                                                                | `true`                                                                                            |
| **attributesToHighlight** | `[Strings]`            | Attributes whose values will contain highlighted matching terms                                 | `true`                                                                                            |
| **matches**               | `Boolean`              | Defines whether an object that contains information about the matches should be returned or not | `true`                                                                                            |

> > `filters` accept a query string. You can find about the filter syntax on [our dedicated page](/guides/advanced_guides/filtering.md).

Reviewer 3: ```suggestion
> `filters` accept a query string. You can find more about the filter syntax on [our dedicated page](/guides/advanced_guides/filtering.md).
```
